src/main/java/com/github/princesslana/eriscasper/ErisCasper.java

```diff
@@ -38,21 +40,23 @@
   private final Payloads payloads = new Payloads(jackson);
 
   private final Routes routes;
+  private final Optional<Shard> shard;
 
-  private ErisCasper(BotToken token) {
-    this.token = token;
+  private Gateway gateway;
```

Comment:
final? </nit>

Reply:
the gateway is initialized later, not when created

Reply:
Is gateway used out side the `getEvents` method?

Reply:
I'll probably remove the field altogether and add it with the action rewrite